msbuild/tests/Xamarin.iOS.Tasks.Tests/FrameworkListTest.cs

```diff
@@ -0,0 +1,147 @@
+			var frameworkListAssemblies = ScanFrameworkListXml (frameworkListFile, isMac);
+			var frameworkName = isMac ? "Xamarin.Mac" : frameworkListFile.Split ('-')[0];
+			var installedAssemblies = ScanAssemblyDirectory (frameworkName, isMac);
+			using (var sw = new StringWriter ()) {
+				using (var writer = XmlWriter.Create (sw, new XmlWriterSettings { Encoding = Encoding.UTF8, ConformanceLevel = ConformanceLevel.Fragment, Indent = true })) {
+					foreach (var assembly in installedAssemblies) {
+						if (!frameworkListAssemblies.Any (a => a.Name == assembly.Name)) {
+							writer.WriteStartElement ("File");
+							writer.WriteAttributeString ("AssemblyName", assembly.Name);
+							WriteNonEmptyAttribute ("Version", assembly.Version);
+							WriteNonEmptyAttribute ("PublicKeyToken", assembly.PublicKeyToken);
+							WriteNonEmptyAttribute ("Culture", assembly.Culture);
+							if (assembly.ProcessorArchitecture != ProcessorArchitecture.None)
+								writer.WriteAttributeString ("ProcessorArchitecture", assembly.ProcessorArchitecture.ToString ());
+							if (assembly.InGac)
+								writer.WriteAttributeString ("InGac", "true");
+							writer.WriteEndElement ();
+							errorCount++;
+						}
+					}
+
+					void WriteNonEmptyAttribute (string name, string val)
+					{
+						if (!string.IsNullOrEmpty (val))
+							writer.WriteAttributeString (name, val);
+					}
+				}
+				Assert.AreEqual (0, errorCount, $"Missing assemblies in '{frameworkListFile}'\n{sw.ToString ()}");
+			}
+		}
+
+		List<AssemblyInfo> ScanFrameworkListXml (string frameworkListFile, bool isMac)
+		{
+			var assemblies = new List<AssemblyInfo> ();
+			var path = string.Empty;
+			if (isMac)
+				path = Path.GetFullPath (Path.Combine ("..", "..", "Xamarin.Mac.Tasks", frameworkListFile));
+			else
+				path = Path.GetFullPath (Path.Combine ("..", "..", "Xamarin.iOS.Tasks.Core", frameworkListFile));
+			using (var reader = XmlReader.Create (path)) {
+				while (reader.Read ()) {
+					if (reader.IsStartElement ()) {
+						switch (reader.LocalName) {
+						case "File":
+							assemblies.Add (ReadFileElement (reader));
+							break;
+						}
+					}
+				}
+			}
+			return assemblies;
+		}
+
+		List<AssemblyInfo> ScanAssemblyDirectory (string frameworkName, bool isMac)
+		{
+			var assemblies = new List<AssemblyInfo> ();
+			var path = string.Empty;
+			if (isMac)
+				path = Path.GetFullPath (Path.Combine (Configuration.SdkRootXM, "lib", "mono", frameworkName));
+			else
+				path = Path.GetFullPath (Path.Combine (Configuration.MonoTouchRootDirectory, "lib", "mono", frameworkName));
+			foreach (var f in Directory.EnumerateFiles (path, "*.dll")) {
+				try {
+					var an = AssemblyName.GetAssemblyName (f);
+					var ainfo = new AssemblyInfo ();
+					ainfo.Update (an);
+					assemblies.Add (ainfo);
+				} catch (Exception ex) {
+					Console.WriteLine ("Error reading assembly '{0}' in framework '{1}':{2}{3}", f, frameworkName, Environment.NewLine, ex);
```

Comment:
might be better to let the test fail in such case... otherwise we'll never realize there's a problem with one of the files